<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Introduction to the Robust longitudinal Differential Expression (RolDE) method</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Introduction to the Robust longitudinal Differential Expression (RolDE) method</h1>


<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#applying-rolde-in-example-datasets-with-aligned-time-points">Applying RolDE in example datasets with aligned time points</a>
<ul>
<li><a href="#semi-simulated-spike-in-dataset">Semi-simulated spike-in dataset</a></li>
</ul></li>
<li><a href="#applying-rolde-in-example-data-with-non-aligned-time-points">Applying RolDE in example data with non aligned time points</a></li>
<li><a href="#changing-the-settings-for-rolde">Changing the settings for RolDE</a></li>
<li><a href="#preparation-of-data-for-rolde">Preparation of data for RolDE</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RolDE)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(printr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Registered S3 method overwritten by &#39;printr&#39;:</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   method                from     </span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   knit_print.data.frame rmarkdown</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span></code></pre></div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>RolDE is a composite method, consisting of three independent modules with different approaches to detecting differential expression. The combination of these diverse modules allows RolDE to robustly detect varying differences in longitudinal trends and expression levels in diverse data types and experimental settings.</p>
<p>The RegROTS module merges the power of regression modelling with the power of the established differential expression method Reproducibility Optimized Test Statistic (ROTS) (Elo et al., Suomi et al.). A polynomial regression model of protein expression over time is fitted separately for each replicate (individual) in each condition. Differential expression between two replicates (individuals) in different conditions is examined by comparing the coefficients of the replicate-specific regression models. If all coefficient differences are zero, no longitudinal differential expression between the two replicates in different conditions exist. For a thorough exploration of differential expression between the conditions, all possible combinations of replicates in different conditions are examined.</p>
<p>In the DiffROTS module the expression of replicates (individuals) in different conditions are directly compared at all time points. Again, if the expression level differences at all time points are zero, no differential expression between the examined replicates (individuals) in different conditions exist. Similarly to the RegROTS module, differential expression is examined between all possible combinations of replicates (individuals) in the different conditions. In non-aligned time point data, the expression level differences between the conditions is examined when accounting for time-associated trends of varying complexity in the data. More specifically, the expression level differences between the conditions are examined when adjusting for increasingly complex time-related expression trends of polynomial degrees d=0,1,.,d where d is the maximum degree for the polynomial and the same degree as is used for the PolyReg module.</p>
<p>In the PolyReg module, polynomial regression modelling is used to detect longitudinal differential expression. Condition is included as a categorical factor within the models and by investigating the condition related intercept and the polynomial termns at different levels of the condition factor, average differences in expression levels as well as differences in longitudinal expression patterns between the conditions can be examined.</p>
<p>Finally, to conclusively detect any differential expression, the findings from the different modules are combined using the rank product. For more details about the method, see the original RolDE publication (Valikangas et al.).</p>
</div>
<div id="applying-rolde-in-example-datasets-with-aligned-time-points" class="section level2">
<h2>Applying RolDE in example datasets with aligned time points</h2>
<p>First, load the dataset and the design matrix to be used:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(data1)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;des_matrix1&quot;</span>)</span></code></pre></div>
<p>To understand the structure of the design matrix and what a design matrix for RolDE should look like, let’s explore the data and the design matrix:</p>
<p>Dimensions of the data:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(data1)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1045   30</span></span></code></pre></div>
<p>Column names of the data:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colnames</span>(data1))</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Condition1_timepoint_1_Sample_1_r1&quot; &quot;Condition1_timepoint_1_Sample_1_r2&quot;</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] &quot;Condition1_timepoint_1_Sample_1_r3&quot; &quot;Condition1_timepoint_2_Sample_2_r1&quot;</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [5] &quot;Condition1_timepoint_2_Sample_2_r2&quot; &quot;Condition1_timepoint_2_Sample_2_r3&quot;</span></span></code></pre></div>
<p>Dimensions of the design matrix:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(des_matrix1)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 30  4</span></span></code></pre></div>
<p>Column names of the design matrix:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(des_matrix1)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Sample Names&quot;         &quot;Condition&quot;            &quot;Time point&quot;          </span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [4] &quot;Replicate/Individual&quot;</span></span></code></pre></div>
<p>Contents of the design matrix:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(des_matrix1)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Sample Names</th>
<th align="left">Condition</th>
<th align="left">Time point</th>
<th align="left">Replicate/Individual</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Condition1_timepoint_1_Sample_1_r1</td>
<td align="left">Condition1</td>
<td align="left">1</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">Condition1_timepoint_1_Sample_1_r2</td>
<td align="left">Condition1</td>
<td align="left">1</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">Condition1_timepoint_1_Sample_1_r3</td>
<td align="left">Condition1</td>
<td align="left">1</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">Condition1_timepoint_2_Sample_2_r1</td>
<td align="left">Condition1</td>
<td align="left">2</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">Condition1_timepoint_2_Sample_2_r2</td>
<td align="left">Condition1</td>
<td align="left">2</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">Condition1_timepoint_2_Sample_2_r3</td>
<td align="left">Condition1</td>
<td align="left">2</td>
<td align="left">3</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(des_matrix1)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Sample Names</th>
<th align="left">Condition</th>
<th align="left">Time point</th>
<th align="left">Replicate/Individual</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">[25,]</td>
<td align="left">Condition2_timepoint_4_Sample_4_r1</td>
<td align="left">Condition2</td>
<td align="left">4</td>
<td align="left">4</td>
</tr>
<tr class="even">
<td align="left">[26,]</td>
<td align="left">Condition2_timepoint_4_Sample_4_r2</td>
<td align="left">Condition2</td>
<td align="left">4</td>
<td align="left">5</td>
</tr>
<tr class="odd">
<td align="left">[27,]</td>
<td align="left">Condition2_timepoint_4_Sample_4_r3</td>
<td align="left">Condition2</td>
<td align="left">4</td>
<td align="left">6</td>
</tr>
<tr class="even">
<td align="left">[28,]</td>
<td align="left">Condition2_timepoint_5_Sample_5_r1</td>
<td align="left">Condition2</td>
<td align="left">5</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">[29,]</td>
<td align="left">Condition2_timepoint_5_Sample_5_r2</td>
<td align="left">Condition2</td>
<td align="left">5</td>
<td align="left">5</td>
</tr>
<tr class="even">
<td align="left">[30,]</td>
<td align="left">Condition2_timepoint_5_Sample_5_r3</td>
<td align="left">Condition2</td>
<td align="left">5</td>
<td align="left">6</td>
</tr>
</tbody>
</table>
<p>The first column of the design matrix should contain the sample names of the data (column names of the data). The second column should indicate the condition/group factor status for each sample to be explored (e.g. sick/healthy, control/case). The third column should indicate the time point for each sample (data with aligned time points) or time value (or equivalent, e.g. age, temperature, disease progression) information in data with non-aligned time points. The fourth and final column of the design matrix should indicate the replicate / individual each sample came from. Let’s look at the exemplary design matrix a little bit more:</p>
<p>Unique samples:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(des_matrix1[,<span class="dv">1</span>])</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;Condition1_timepoint_1_Sample_1_r1&quot; &quot;Condition1_timepoint_1_Sample_1_r2&quot;</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] &quot;Condition1_timepoint_1_Sample_1_r3&quot; &quot;Condition1_timepoint_2_Sample_2_r1&quot;</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;Condition1_timepoint_2_Sample_2_r2&quot; &quot;Condition1_timepoint_2_Sample_2_r3&quot;</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;Condition1_timepoint_3_Sample_3_r1&quot; &quot;Condition1_timepoint_3_Sample_3_r2&quot;</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;Condition1_timepoint_3_Sample_3_r3&quot; &quot;Condition1_timepoint_4_Sample_4_r1&quot;</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] &quot;Condition1_timepoint_4_Sample_4_r2&quot; &quot;Condition1_timepoint_4_Sample_4_r3&quot;</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] &quot;Condition1_timepoint_5_Sample_5_r1&quot; &quot;Condition1_timepoint_5_Sample_5_r2&quot;</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] &quot;Condition1_timepoint_5_Sample_5_r3&quot; &quot;Condition2_timepoint_1_Sample_1_r1&quot;</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;Condition2_timepoint_1_Sample_1_r2&quot; &quot;Condition2_timepoint_1_Sample_1_r3&quot;</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] &quot;Condition2_timepoint_2_Sample_2_r1&quot; &quot;Condition2_timepoint_2_Sample_2_r2&quot;</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] &quot;Condition2_timepoint_2_Sample_2_r3&quot; &quot;Condition2_timepoint_3_Sample_3_r1&quot;</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [23] &quot;Condition2_timepoint_3_Sample_3_r2&quot; &quot;Condition2_timepoint_3_Sample_3_r3&quot;</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] &quot;Condition2_timepoint_4_Sample_4_r1&quot; &quot;Condition2_timepoint_4_Sample_4_r2&quot;</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [27] &quot;Condition2_timepoint_4_Sample_4_r3&quot; &quot;Condition2_timepoint_5_Sample_5_r1&quot;</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [29] &quot;Condition2_timepoint_5_Sample_5_r2&quot; &quot;Condition2_timepoint_5_Sample_5_r3&quot;</span></span></code></pre></div>
<p>Conditions:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(des_matrix1[,<span class="dv">2</span>])</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Condition1&quot; &quot;Condition2&quot;</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(des_matrix1[,<span class="dv">2</span>])</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">Condition1</th>
<th align="right">Condition2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">15</td>
<td align="right">15</td>
</tr>
</tbody>
</table>
<p>Time points:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(des_matrix1[,<span class="dv">3</span>])</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; &quot;5&quot;</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(des_matrix1[,<span class="dv">3</span>])</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
<th align="right">5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">6</td>
<td align="right">6</td>
<td align="right">6</td>
<td align="right">6</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
<p>Replicates:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(des_matrix1[,<span class="dv">4</span>])</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; &quot;5&quot; &quot;6&quot;</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(des_matrix1[,<span class="dv">4</span>])</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
<th align="right">5</th>
<th align="right">6</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
</tr>
</tbody>
</table>
<p>In this example, we have a dataset of 30 samples, 2 conditions, 6 replicates each with 5 time points. The timepoints in the data are aligned.</p>
<p>This is how a design matrix for a dataset for RolDE should look like. This gives all the essential information for RolDE it needs to determine longitudinal differential expression between the conditions.</p>
<p>By bare minimum, RolDE needs the data and the design matrix. By default RolDE assumes that the time points in the data are aligned. If not defined otherwise, RolDE will by default use sequential processing. However, using parallel processing and multiple threads will greatly reduce the computational time required by RolDE.</p>
<p>Please use set.seed() for reproducibility.</p>
<p>Running RolDE using parallel processing and 3 threads:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>data1.res<span class="ot">&lt;-</span><span class="fu">RolDE</span>(<span class="at">data=</span>data1, <span class="at">des_matrix=</span>des_matrix1, <span class="at">n_cores=</span><span class="dv">3</span>)</span></code></pre></div>
<p>RolDE supports the SummarizedExperiment data structure and the data and design matrix can be provided to RolDE as a SummarizedExperiment object. In this case, the <em>data</em> argument for RolDE must be a SummarizedExperiment object, where the data matrix is included as a list in the <em>assays</em> argument and the design matrix in the <em>colData</em> argument. The format of the data matrix and the design matrix within the SummarizedExperiment object must be the same as when provided separately.</p>
<p>Constructing a SummarizedExperiment object from data1 and the associated design matrix for RolDE:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>SE_object_for_RolDE <span class="ot">=</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays=</span><span class="fu">list</span>(data1),</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">colData =</span> des_matrix1)</span></code></pre></div>
<p>Running RolDE using a SummarizedExperiment object including the data and the metadata:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>data1.res<span class="ot">&lt;-</span><span class="fu">RolDE</span>(<span class="at">data=</span>SE_object_for_RolDE, <span class="at">n_cores=</span><span class="dv">3</span>)</span></code></pre></div>
<p>The results of RolDE are returned in a list with lot of information:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(data1.res)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;RolDE_Results&quot;     &quot;RegROTS_Results&quot;   &quot;RegROTS_P_Values&quot; </span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] &quot;DiffROTS_Results&quot;  &quot;DiffROTS_P_Values&quot; &quot;PolyReg_Results&quot;  </span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;PolyReg_P_Values&quot;  &quot;ROTS_Runs&quot;         &quot;Method_Degrees&quot;   </span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] &quot;Input&quot;</span></span></code></pre></div>
<p>The main results of RolDE are located in the first element of the provided result list. Elements 2,4 and 6 provide the result for the different modules of RolDE separately (the RegROTS, DiffROTS and PolyReg modules).</p>
<p>Elements 3 and 5 provide the significance values for the different ROTS runs of the RegROTS and DiffROTS modules, respectively. The used ROTS runs within those modules are given in element 8. The comparison between replicates in different conditions are divided into the different runs so that each sample is used only once within each run to preserve the proper degrees of freedom for statistical testing.</p>
<p>All the condition related significance values of the polynomial regression models in the PolyReg module are given in element 7. The used polynomial degrees for the regression models in the RegROTS and the PolyReg modules are given in element 9. And in element 10, all the inputs used by RolDE (both given by the user and those determined automatically by the method) are given.</p>
<p>Typically, the main (only) interest of the user is in element 1, where the main results of RolDE are given:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>RolDE.data1<span class="ot">&lt;-</span>data1.res<span class="sc">$</span>RolDE_Results</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(RolDE.data1)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1045    4</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(RolDE.data1)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Feature ID&quot;                           </span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [2] &quot;RolDE Rank Product&quot;                   </span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] &quot;Estimated Significance Value&quot;         </span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [4] &quot;Adjusted Estimated Significance Value&quot;</span></span></code></pre></div>
<p>By default, the features in the result data frame are given in the same order as entered.</p>
<p>Let’s order the results based on the strength of longitudinal differential expression detected by RolDE:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>RolDE.data1<span class="ot">&lt;-</span>RolDE.data1[<span class="fu">order</span>(<span class="fu">as.numeric</span>(RolDE.data1[,<span class="dv">2</span>])),]</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(RolDE.data1, <span class="dv">5</span>)</span></code></pre></div>
<table>
<colgroup>
<col width="21%" />
<col width="21%" />
<col width="12%" />
<col width="19%" />
<col width="25%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Feature ID</th>
<th align="right">RolDE Rank Product</th>
<th align="right">Estimated Significance Value</th>
<th align="right">Adjusted Estimated Significance Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">sp|P32263|P5CR_YEAST</td>
<td align="left">sp|P32263|P5CR_YEAST</td>
<td align="right">4.447960</td>
<td align="right">0.0010988</td>
<td align="right">0.9983582</td>
</tr>
<tr class="even">
<td align="left">P01344ups|IGF2_HUMAN_UPS</td>
<td align="left">P01344ups|IGF2_HUMAN_UPS</td>
<td align="right">6.073178</td>
<td align="right">0.0028508</td>
<td align="right">0.9983582</td>
</tr>
<tr class="odd">
<td align="left">sp|Q04935|COX20_YEAST</td>
<td align="left">sp|Q04935|COX20_YEAST</td>
<td align="right">7.989570</td>
<td align="right">0.0044291</td>
<td align="right">0.9983582</td>
</tr>
<tr class="even">
<td align="left">sp|P40008|FMP52_YEAST</td>
<td align="left">sp|P40008|FMP52_YEAST</td>
<td align="right">9.356095</td>
<td align="right">0.0053161</td>
<td align="right">0.9983582</td>
</tr>
<tr class="odd">
<td align="left">sp|P15180|SYKC_YEAST</td>
<td align="left">sp|P15180|SYKC_YEAST</td>
<td align="right">9.619002</td>
<td align="right">0.0061172</td>
<td align="right">0.9983582</td>
</tr>
</tbody>
</table>
<p>Explore the distribution of the estimated significance values:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(RolDE.data1<span class="sc">$</span><span class="st">`</span><span class="at">Estimated Significance Value</span><span class="st">`</span>, <span class="at">main =</span> <span class="st">&quot;Estimated significance values&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAqAAAAHgCAMAAABNUi8GAAAAbFBMVEUAAAAAADoAAGYAOjoAOpAAZpAAZrY6AAA6ADo6AGY6OgA6kLY6kNtmAABmADpmtv+QOgCQOjqQZgCQkGaQ29uQ2/+2ZgC2Zma2/7a2/9u2///T09PbkDrb/7bb////tmb/25D//7b//9v///8BDOKRAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAUcklEQVR4nO2d63qjSpJF5ZpqVc+M3fZcStNTTZfK1vu/Y0sQSIktybpAsMBr/zifHWZnBME6CaRUsNgoBdZi7AKUOicBVWgJqEJLQBVaAqrQElCFloAqtARUoSWgCi0BVWgJqEJLQBVaAqrQElCFloAqtARUoSWgCi0BVWgJqEJLQBVaAqrQElCFloAqtARUoSWgCi0BVWgJqEJLQBVaAqrQElCFloAqtARUoTVjQKtFq++/u3/5v1+bt5cP0Q/abnZEq8W3D/Gzo8UfV9tC/nJB1lt0yd5MVF8Q0Nenb5cAWm92RDcCut4V8p8Ceq2+IKDHCDuiU5tdaP9YzcPPG2yXSUCnqKpDUtWSujvVLr79/+6Qvj4tlruZ7Xn314afPz8WzY/NZr+aH5bNGKvdnwpA92O2gBw2qOof6pHqP9bDPfxvbFdPp8+dfKWjs0VZwO6XeoOt7bEwN/mbyrY79di1VR//J52OvgqgMZ1uI11AC+22Xjc/bg96ABqb7P62peCwYXfMBpBmg4f/aADdj/QR0FXzt8cyX+kotygK2GndYFvPx+tuig6ghe1Q5hQ1a0BbPe+O17I+uo9xjn5rAX3c7OfQx1102U5Pq5g/t/9tjvm6npFW+0NdjLm/zNxvULX/bf9YHUCqLVv3t19lvsJRbFEWcEhaj1OYPwJ6sJW7PkF9EUC3h3E/g3QA3R7X5r8NJa1v2UGzmYz3xz9GKsZ8B0gD2nO9yfIjoM12VTkRLzelo9iiLCC23v70vtgPgBa2ctcnqC8CaJye4+KsAHTZTkrNMW9P4y0k9VVecx5tJqziJqkYs53R4kI0rkFj6PeAlnc0Rb4PjlpFAfvA83Y+LK85jgDarXu/6xPUrAEtZo72cvP5LKD1wXxuAvVm68X+QDfR8i7+MGYx2kWAtvc8Zb6uo92iKGBvWW5TLDvmEtB6R0pbsesT1FcBdHO4mT0DaDONFoDW81WjzgT5bsxbZ9Ay38kZ9B1Y2936+1NcDTx2DAWg72wTvo//KoC+/Vdzc9xeSx4HdL04HPn2GrRY4elegxZjHrsGPQlobLdu7sOfDzdJe0exRVlAo9enh/9pVxyeOzdJzQjtDdbyyK5PULMGtFV9p9CsA3VQOTaD1jPcYUZbxRrAu5v0nYoxj93FnwZ0HZeNi+eP+fb33LFFWUCoXeAszG2KuEp5LOsud32C+iKAtr/EFFMs1H+8Bo2blnqzdh20XJx8vw66B6+7Dnoa0HagZSdfCehhi01ZQKN1/FaYm3H3d0aPHVux6xPUVwG0OXoNWqt2yfz4XfzDz6r9CKlhan/5tsPmuThX7scsPkn69uuza9BN3PrE7X+brwPoYYtOAbX20+nBHOPu6vn+z6f9yLFdsevT04wBHUuriZ5MmRLQ/rRaxAfky8+3VRdKQPtTu/g40as9pgS0RzV3Jp7g+5SAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKqrtHiFt2VsK/K1ZfQ4h/XS0BVmgRUoSWgCi0BVWgJaKPse0V1oQS00Q19gO7JzCSgjbCApq8DDqlb9kVAa3EBzT5AQ+qWLmfvP7R5ApohAb1ZOYBO4RQH2xkBbZQE6A1Zrrdw5/Yp7L+AXpnleouA3rVbV2399tKcGr79uifnBRJQ6M6wAa0Wj80P6/aHoSSg0J1BA/r2ssey+v77nqyfal6A3iDqzqABfX16bn9cD3ySnxegN1ioO4MG1Bk0D1DqmhEa0O01aEyhXoMO7JmTJfMu/vWp+Z914PlTQOdkcR305jYIaIZFQG9ug4BmWFyojzZQ7yu46MwOUPZCPdXCrWxugMKXmagWbmVzA/T0Qv09n4AcLyuldQKaYUHNoAI6WhqsBbVQL6CjpcFaUAv1AjpaGqwFtQ4qoKOlwVoE9ObWCWiGRUBvbp2AZlgE9ObWCWiGJXMd9LDeeeLDTgEdLQ3WkjiDvr189iG8gI6WBmvJ/bLIss/hzoyT0joBzbCkXoOuF89n/y6go6XBWrxJurl1ApphEdCbWyegGRYBvbl1ApphEdCbWyegGRYBvbl1ApphEdCbWyegGRY+oNh/ziagGZYJAJrSB6yFW5mARjClD1gLtzIBjWBKH7AWbmUCGsGUPmAt3MoENIIpfcBauJUJaART+oC1cCsT0Aim9AFr4VYmoBFM6QPWwq1MQCOY0geshVuZgEYwpQ9YC7cyAY1gSh+wFm5lAhrBlD5gLdzKBDSCKX3AWriVCWgEU/qAtXArE9AIpvQBa+FWJqARTOkD1sKtTEAjmNIHrIVbmYBGMKUPWAu3MgGNYEofsBZuZQIawZQ+YC3cygQ0gil9wFq4lQloBFP6gLVwKxPQCKb0AWvhViagEUzpA9bCrWx+gH76tmMBHS0N1oJ627GAjpYGa+G/qxPbOgHNsBDednx+OGzrBDTD4gwKt3Armxugt77tGNs6Ac2w8N92jG2dgGZYXAeFW7iVfR1ADw+ePfrXlD5gLdzK5gfoarFY1ov1p17nJaCjpcFaMm+Stteeq8Vydy3qTRItDdaSvMy0fvi5cZkJmAZrSV6ob5boXajHpcFanEHhFm5lcwP0cA1afKZ0wXDY1glohsW7eLiFW9n8AL1tOGzrBDTDIqBwC7cyAY1gSh+wFm5lAhrBlD5gLdzKBDSCKX3AWriVCWgEU/qAtXArE9AIpvQBa+FWJqARTOkD1sKtTEAjmNIHrIVbmYBGMKUPWAu3MgGNYEofsBZuZQIawZQ+YC3cygQ0gil9wFq4lQloBFP6gLVwKxPQCKb0AWvhViagEUzpA9bCrUxAI5jSB6yFW5mARjClD1gLtzIBjWBKH7AWbmWTAvT1afev4e6VgI6WBmvpbQatFotTT1W8ZbgimNIHrIVb2cQA3fTAqICOlgZr6fcatDr3Do/rh2uCKX3AWriVTQ7Qdf0v3t9eTj6f9rrh9sGUPmAt3MqmBeju6ckNmacevHTVcGUwpQ9YC7eySQH6+lQ/dOlOCehoabAW10HhFm5lEwN0tT3Bn3x9x/XDFcGUPmAt3MqmBeiqvgB9fbpruV5AR0uDtfR2Ddo8se6OO6SNgI6YBmvpCdD2kZ+VgPZs4VY2KUDjNXJ/fvhJUs8WbmXTAnTL5mKxuHOtSUBHS4O1uMwEt3ArE9AIpvQBa+FWNi1A62fPf/JVkU+3EdDR0mAtva2Dfn77XrXL+L6Om5cGa+ltHfTT2/fi5TO+JwmXBmvpeaH+jIpNfNMcLg3W0ttC/aefcTqDCuiI16Drk2/n2qtqN/EalJcGa+nvX3V+fhffbnTyK/cCOloarMV1ULiFW9nXAXSx19G/pvQBa+FWNjFAt+fv779XZxebdmulu4/sT35iL6CjpcFaertJevi5vTc/uxxa8/nXn2fWpAR0tDRYS4/fB90tHp35PmgN76pejXKZCZcGa+lxoX6H3Zlv1O82ibVQF+pxabCWnmfQ1ZmnNuxmz8oZlJkGa+n3GrQ6t1z/+vTtVz2Frk/dJQnoaGmwlj7v4j/9Rv26WUw6+amogI6WBmshrIOeHw7bOgHNsAgo3MKtbFKAXvRZ/OXDdYIpfcBauJVNCtDQfc9tENDx0mAtPZ/iVz76pmcLt7JJAuqjb/q2cCubJKA++qZvC7eyKQLq0+16t3ArmxSgn35Z/rrhOsGUPmAt3MomBWg/EtDR0mAtAgq3cCubFKD7hfq71uoFdLQ0WEu//+z47LeZrhmuDKb0AWvhVjYpQN9eGjJPfdPzyuE6wZQ+YC3cyiYF6Ovfmi/auVDft4Vb2aQAbWfQc9+ov2K4TjClD1gLt7JJARrfkq/uewa4gI6WBmvp9xv1963TC+h4abAW10HhFm5lAhrBlD5gLdzKJgboBY++uWa4IpjSB6yFW9m0AL3g0TfXDFcGU/qAtXArmxSgFzz65prhOsGUPmAt3MomBegFj765ZrhOMKUPWAu3skkBesmjb64YrhNM6QPWwq1sUoBe8uiba4Yrgyl9wFq4lU0L0IsefXPFcEUwpQ9YC7eyiQHahwR0tDRYS4/XoPdLQEdLg7XkvWnumuE6wZQ+YC3cyiYF6L0PvXk/XBlM6QPWwq1sUoD68LChLNzKJgVoPxLQ0dJgLQIKt3Armw6g/dwhbQR0xDRYS2+AXrDQ9PbyyXWqgI6WBmtJBLRq38Lt67h5abCWPECLDXxPEi4N1pIHaHGl6pvmcGmwFmdQuIVb2dwAPTy4yWtQXhqspR9AL3u03adPuRXQ0dJgLS7Uwy3cyr4OoIf59+hfU/qAtXArmyugf36c/NxJQEdLg7XkAXrBlaqAjpYGa0mcQePm3RmUmAZryTzFvz7t7t8FlJgGa8m9Bl09/BRQZBqsJfkmqVo8CigxDdaSfRf/58e/CSgwDdaSvsz09nL66SMCOloarIWwUH9+OGzrBDTDIqBwC7cyAY1gSh+wFm5lAhrBlD5gLdzKBDSCKX3AWriVCWgEU/qAtXArE9AIpvQBa+FWJqARTOkD1sKtTEAjmNIHrIVbmYBGMKUPWAu3MgGNYEofsBZuZQIawZQ+YC3cygQ0gil9wFq4lQloBFP6gLVwKxPQCKb0AWvhViagEUzpA9bCrUxAI5jSB6yFW5mARjClD1gLtzIBjWBKH7AWbmUCGsGUPmAt3MoENIIpfcBauJUJaART+oC1cCsT0Aim9AFr4VYmoBFM6QPWwq1MQCOY0geshVuZgEYwpQ9YC7cyAY1gSh+wFm5lAhrBlD5gLdzKBDSCKX3AWriVCWgEU/qAtXArE9AIpvQBa+FWJqARTOkD1sKtTEAjmNIHrIVbmYBGMKUPWAu3svkB+vbyyQtnBXS0NFhLIqBV+xZuX8fNS4O15AFavFC+OvFCbgEdLQ3Wkvmuzv3bPda+q5OWBmtxBoVbuJXNDdDtNWhMoV6D8tJgLbkvk23u4k/MnwI6YhqsxXVQuIVb2dcBdLHX0b+m9AFr4VY2P0CrLYL1ZWjlXTwtDdaSeZP08HN7GbrcCCgwDdaSvcz09rK9RRJQXBqsJX+hfvX9t4Di0mAtIyzUr5YCikuDtWRegwaWr0+nvs8koKOlwVpy7+Kbk/zbi4DS0mAthHXQ88NhWyegGRYBhVu4lQloBFP6gLVwKxPQCKb0AWvhViagEUzpA9bCrUxAI5jSB6yFW5mARjClD1gLtzIBjWBKH7AWbmUCGsGUPmAt3MoENIIpfcBauJUJaART+oC1cCsT0Aim9AFr4VYmoBFM6QPWwq1MQCOY0geshVuZgEYwpQ9YC7cyAY1gSh+wFm5lAhrBlD5gLdzKBDSCKX3AWriVCWgEU/qAtXArE9AIpvQBa+FWJqARTOkD1sKtTEAjmNIHrIVbmYBGMKUPWAu3MgGNYEofsBZuZQIawZQ+YC3cygQ0gil9wFq4lQloBFP6gLVwKxPQCKb0AWvhViagEUzpA9bCrUxAI5jSB6yFW5mARjClD1gLtzIBjWBKH7AWbmUCGsGUPmAt3MoENIIpfcBauJUJaART+oC1cCsT0Aim9AFr4VYmoBFM6QPWwq1MQCOY0geshVuZgEYwpQ9YC7cyAY1gSh+wFm5l8wP07aV5K/yJ13gJ6IhpsJbMVyEu4mWd6/aHi4bDtk5AMywjvEx2i+r335cPh22dgGZY8l/HvdW6e5Jf7HU0h/raupHNBp4rtr1gBlWqX115DRpT6MlrUKX61XXT7+tTM2c7f6ok9bwOqlS/ElCFloAqtARUoSWgCi0BVWgJqEJLQBVaGYCO+0GwGl13wdMXhSPnmFcWd6YfMyjHvLK4M/2YQTnmlcWd6ccMyjGvLO5MP2ZQjnllcWf6MYNyzCuLO9OPGZRjXlncmX7MoBzzyuLO9GMG5ZhXFnemH7NSQ0tAFVoCqtASUIWWgCq0BFShJaAKLQFVaAmoQktAFVoCqtASUIWWgCq0BgR0vVg8/Dz6y1BZ6peQDPJo3fflr4Z5QmonzZ8fi8Vy8CzVtmXP57a+Q3/+ff+g+JuP/3CArrcFrduiOr8MleXtZftDNcRBfV/+ephH+HZbtk3x+jT0zlS7XwYi9PVp/yaD24//YIA2D7RfLT/+MliWPz92ja5OvsSpnyyb+kHTQwB6pGVD78zby3IzyIHZ1JNmW/0dx38wQDu0DIbOkYEHmKjfZ6m+//cQgHZb9tdBrojeZRkQ0PXicf8umDuO/3CA1g1el91eDwDox4FXg2fZ/jrINWgnzfrb358GuaDu7syQp/hOxzY3Hv/BAG0mspjOOr8MlqWJDHBQu1l256tBAO2kqXbnx2aCGzDLgHevm4LIO47/3ABdD3WPdMiye0dUAqAPA510ujuzO9/8+THQS4XQgI5zih/mBU4f92X4U3xzwdZcvA2XZbCbg1roU/woN0nVMKugnSxVPFOw/+u2TprmaA5wq9TNMtSprRb6JmmEZabDm/AGzbLTIDNoJ03zYtQBTjpHVuYGyFJrPy5xmWmEhfrBrqU+lj/MJ0ndJfRtiuLtqANlybkGJS7U1+fCXUnNnWg11M1ikSVOvkPk6ezLZrCPOjtp1kN9btvJshoqyyYAvfP4+2URhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWgKq0BJQhZaAKrQEVKEloAotAVVoCahCS0AVWv8C8A0PetXh6yAAAAAASUVORK5CYII=" /><!-- --></p>
<p>As can be observed, the distribution of the estimated significance values is approximately uniform. Data1 is random data; the values for the features have been randomly assigned from a normal distribution with a mean of 22 and a standard deviation of 1.5. Overall, the null hypothesis between Condition1 and Condition2 is true; the conditions are not differentially expressed and a uniform significance value distribution is expected.</p>
<p>After correcting for the simultaneous testing of multiple hypothesis by using the Bejamini-Hochberg (FDR) correction, no differentially epxressed feature remains with the commonly used FDR level of 0.05:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(RolDE.data1<span class="sc">$</span><span class="st">`</span><span class="at">Adjusted Estimated Significance Value</span><span class="st">`</span><span class="sc">&lt;=</span><span class="fl">0.05</span>))</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<p>The calculation of significance values in RolDE is controlled via the parameter <em>sigValSampN</em> which control if the significance values should be calculated and how many permutations should be used when determining the significance values. Computational time required by RolDE can be reduced by not calculating the significance values by setting <em>sigValSampN</em> to 0. By increasing <em>sigValSampN</em> from the default number, the significance values can be estimated more accurately but the required computational time will also increase. If parallel processing for RolDE is enabled (<em>n_cores</em> &gt; 1), it will also be utilized when estimating the significance values.</p>
<p>The estimated significance values can be adjusted by any method included in the p.adjust method in the stats package. Alternatively, q-values as defined by Storey et al. in the Bioconductor package qvalue can be used. Valid values for <em>sig_adj_meth</em> are then: “holm”, “hochberg”, “hommel”, “bonferroni”, “BH”, “BY”,“fdr”,“none”, “qvalue”.</p>
<div id="semi-simulated-spike-in-dataset" class="section level3">
<h3>Semi-simulated spike-in dataset</h3>
<p>In addition to the random data already explored, a semi-simulated spike-in dataset with differential expression between the conditions is also included in the RolDE package:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;data3&quot;</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;des_matrix3&quot;</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>?data3</span></code></pre></div>
<div class="r-help-page">
<div class="container">
<table width="100%" summary="page for data3">
<tr>
<td>
data3
</td>
<td style="text-align: right;">
R Documentation
</td>
</tr>
</table>
<h2>
A semi-simulated UPS1 spike-in dataset with differences in longitudinal expression for the spike-in proteins - no missing values.
</h2>
<h3>
Description
</h3>
<p>
A longitudinal proteomics dataset with five timepoints, two conditions and three replicates for each sample at each timepoint in each condition. The expression values for each protein in each sample have been generated by using the <code>rnorm</code> function and the means and standard deviations of the corresponding proteins and samples in the original experimental UPS1 spike-in data. In this manner, the expression for each protein in each sample could be replicated with some random variation multiple times when necessary. The pattern of missing values was directly copied from the used samples in the original UPS1 spike-in dataset. All proteins with missing values have been filtered out. Linear-like trends for the spike-in proteins for Condition 1 were generated using the 4,4,10,25 and 50 fmol samples of the original UPS1 spike-in data. Linear-like trends for the spike-in proteins for Condition 2 were generated using the 2,4,4,10 and 25 fmol samples of the original UPS1 spike-in data. For more information about the generation of the semi-simulated spike-in datasets, see the original RolDE publication. In the spike-in datasets, the UPS1 spike-in proteins are expected to differ between conditions while the expression of the rest of the proteins (the background proteins) are expected to remain stable between the conditions, excluding experimental noise.
</p>
<h3>
Usage
</h3>
<pre class="r">
data3
</pre>
<h3>
Format
</h3>
<p>
A matrix with 1033 rows and 30 variables.
</p>
<h3>
Source
</h3>
<p>
<a href="https://www.ebi.ac.uk/pride/archive/projects/PXD002099">https://www.ebi.ac.uk/pride/archive/projects/PXD002099</a>
</p>
</div>
</div>
<p>Let’s run RolDE for data3 using the Bonferroni procedure for multiple hypothesis adjustement:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>data3.res<span class="ot">&lt;-</span><span class="fu">RolDE</span>(<span class="at">data=</span>data3, <span class="at">des_matrix=</span>des_matrix3, <span class="at">n_cores=</span><span class="dv">3</span>, <span class="at">sig_adj_meth =</span> <span class="st">&quot;bonferroni&quot;</span>)</span></code></pre></div>
<p>Retrieve the final RolDE results and organize the results based on strength of differential expression:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>RolDE.data3<span class="ot">&lt;-</span>data3.res<span class="sc">$</span>RolDE_Results</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>RolDE.data3<span class="ot">&lt;-</span>RolDE.data3[<span class="fu">order</span>(<span class="fu">as.numeric</span>(RolDE.data3[,<span class="dv">2</span>])),]</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(RolDE.data3, <span class="dv">3</span>)</span></code></pre></div>
<table>
<colgroup>
<col width="20%" />
<col width="20%" />
<col width="13%" />
<col width="19%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Feature ID</th>
<th align="right">RolDE Rank Product</th>
<th align="right">Estimated Significance Value</th>
<th align="right">Adjusted Estimated Significance Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">P41159ups|LEP_HUMAN_UPS</td>
<td align="left">P41159ups|LEP_HUMAN_UPS</td>
<td align="right">5.235409</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P01133ups|EGF_HUMAN_UPS</td>
<td align="left">P01133ups|EGF_HUMAN_UPS</td>
<td align="right">5.473704</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">P01112ups|RASH_HUMAN_UPS</td>
<td align="left">P01112ups|RASH_HUMAN_UPS</td>
<td align="right">7.894447</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Explore the distribution of the estimated significance values:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(RolDE.data3<span class="sc">$</span><span class="st">`</span><span class="at">Estimated Significance Value</span><span class="st">`</span>, <span class="at">main =</span> <span class="st">&quot;Estimated significance values&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAqAAAAHgCAMAAABNUi8GAAAAb1BMVEUAAAAAADoAAGYAOpAAZpAAZrY6AAA6ADo6AGY6OgA6OpA6kLY6kNtmAABmADpmtv+QOgCQOjqQZgCQkGaQ29uQ2/+2ZgC2Zma2/7a2/9u2///T09PbkDrb2//b/7bb////tmb/25D//7b//9v///8StKzFAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAQJUlEQVR4nO2da3ujuhVGSVLP6cWu3cu4pyntkNj//zcWkMCSMbYlJOW1s9aHeRyFLQmxRqBtAtURQJjqqzsAcA0EBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQ5okFrauBt1/+b/71fjzsJqUT2s0usK9eJ+VXa7O/3Lcd+cMdrcZwz948KN9Q0M/N6z2C9ptdIFLQpuvIXxA0lG8o6CXDLjC32Z3h0968/IwIuw8EfURqz6R6MLU71Vav/+4O6eemWnUz27b7rfHn40dlPprN3s2Hlalj3/3KEXSscxDktEHdf+hr6n/ZV/fyT7tdP51uvfbcCG8LtwPdD/0GbdjaCTbtm561O7X2w+rpf9LH4bsIaqfTtsQX1KHbujEf24NuBbWbdL9rLTht6NdpBDEbvPzJCDrWNBV0b363dttzI9wtnA50NEbbfj5u/CY8QZ2wUzcfkacWdGDbHa9Vf3TX9hx9GARdH8c5dN2VrobpaW/nz/Zfc8ybfkbaj4faqXO8zBw3qId/h1/WJ5H6kDb69d1tz4lwtnA7cGq0r8cJngp6CnN3/QH5JoK2h3GcQTxB2+Nq/jWWDHErT00zGY/H39bk1HkmiBFt22+ymgpqtqvdiXh1dCOcLdwO2K3bT+ednQjqhLm7/oB8E0Ht6dlenDmCroZJyRzz4TQ+SNJf5ZnzqJmwnEWSU+cwo9kLUXsNaqs+F9Rd0TjtTSJ6nA6MBdt2PnSvOS4I6vd73PUH5KkFdWaO4XJze1XQ/mBuTUG/WVONB9qUuqv4U51ObXcJOqx53Pb8iGELpwNjyKptYuUFu4L2O+KGObv+gHwXQY+nxewVQc006gjaz1cGb4I8qzN2BnXbm51Bz8Rqd+v3jb0aWHsBjqBnYQ+8jv8ugh7+ZhbHw7XkZUGb6nTkh2tQJ8PjX4M6dV66Bp0V1G7XmHX49rRIGiOcLdwOGD43L/8YMg5bb5FkahgWWKsLu/6APLWgA/1KweSBPFUuzaD9DHea0fY2B3C2SO9w6ry0ip8XtLGXjdV22t645rZbuB2wDAlOJ3howl6lrN1+u7v+gHwTQYcf7BTjJOqn16B20dJvNuRB3eTkeR50FM/Pg84LOlS08tpzBT1tcXQ7YGjsT06wqXdcGa29MGfXH5DvIqg5ekat/ZAyv7yKf/lZD18hGafGy7dOm61zrhzrdL5Jen2/dQ16tEsfu/wf2vMEPW3hdaBnnE5Pwbberj9v/9uMNdvtnF1/PJ5Y0K9i/6AnU00QNB37yn5Bvrq9LdwJgqZjSD4+6NWeJgiaELMy4QSfEgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpCkhaBVBgW7BI1BE0P8Eg6BgQFCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpAkz4bAz74F7fQ9qA0EhliAT6mptPjTDh/vaQFCIJcSEw27Usn77FdAGgkIsISZ8brbDxybkJI+gEA0zKEgTeA1qp1CuQaEQYSZ8bswqPmT+RFBYAHlQkAZBQRoS9SANiXqQhjQTSJMmUV+NXGwDQSGWxDMogkJaEifqERTSkjhRj6CQlsQmICikBUFBmggTmqp6+RlSHYJCNGEm7Ktq/fHnX27C6Y7qEBSiCTJh3y6O9v3sSZoJyhCcqP/4rRN07o56BIW0hAnaZT8P/z0yg0IpwhL1w7xpVL23OgSFaMJMqM3yvalm1kgICokhDwrSIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSjCZ8bqpVwuq8QgSFWBwT6qqqZl5/FFOdU4igEItvwmJHERTScm5Cp+jMezhjqjOFCAqxeCY0rZ3b42E38yLO0OrGQgSFWE4mtKukypg59yrjoOrcQgSFWJxVvHkPZ6LqvEIEhVjIg4I0jgn79gTfLEw0ISik5WTCvr8A/dwsStcjKKTFuQY174BfsEI6IiikZjThsDMn9xpBQYiTCXXVTaEfP/gmCYRwTPj4UVXVwlwTgkJawkw47Kqe2csABIW0BJlQD0mo2WwUgkJaTibcnB3HdVRLPfN1PYJCWpw86M3l+5CJOs4noxAU0uLkQW8u35lBoTiTRP01TCbqyDUoFMNJ1N/xHWd3S17H7A2jCAppOZnQVLen0IDq3EIEhVjcv+q8tYoPqc4rRFCIhUQ9SEOiHqRxTGhP8m+/9leSTaSZoDjOIunlZ6vdtXTofKK+GrnYBoJCLN79oN28eOV+UGZQKI6XqO+0u3ZHPYl6KM1kBt1fe2pDuUR9BIvGAUQ5vwatl6XrkwkaEbKk36CKv4rXuaMeQaEn7LDW/cObjvN/WoegkJawRH07v5o/nEdQKEPId/EmzdQ//A5BoQznh/ValmlI1LcLfQSFMkwO637+ttAxUb9fISiUYXJYrybq7e/aywEEhSJMDuvVR98MWdLDDkGhCOeHVeXpdggKPZNVfPzz6b3qvEIEhVhUn7CMoNCDoCDNNFG/6O/mEBTScv5nx9zNBFI494MaM+fulQ+szitEUIjldIr/q7nRTuQZ9QgKPZMZ9Ood9fdX5xUiKMTi3VF/NHfUJanOLURQiOX8jvpleXoEhcSQBwVpEBSkCXn0TVh1TiGCQiwhj74Jqs4tRFCIJeTRNyHVeYUICrEEPfomoDqvEEEhlrBH39xfnVeIoBALj74BaXj0DUhDHhSk8a5BE1bnFSIoxBL0prmA6rxCBIVYnEXSovzSpDq3EEEhFtUXeSEo9LBIAmkQFKQxhzXNCumIoJAaR9AUiSYEhbQgKEiDoCANgoI0CArSIChIMwia4tF2RwSF1JCoB2kQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQRoEBWkQFKRBUJAm7LAedjfuGUVQSEvQYa0re899U83cfI+gkJaQw+r8TcjcO5ERFNISclid54/MPQoPQSEtzKAgTeA1qJ1CuQaFQoQd1uGvP2dfVYOgkBbyoCANgoI0JOpBGhL1IA1pJpAmTaL+9GSni20gKMTCDArSkKgHaZ4oUR9B6GhBcZ4oDxoRknbnIQMICtIgKEiDoCBNWB705pPsERTSEnSMDrtbb1hAUEhL6M0iq4jqEBSiCTxGTXX9tcgICmlhkQTSIChIg6AgzfcWlK/v5fnegkaEpB0vuAWCBoakHS+4BYIGhqQdL7gFggaGpB0vuAWCBoakHS+4BYIGhqQdL7gFggaGkJoqC4JmD2HWXQKCZg9B0CUgaPYQvrBaAoJmD2EttgQEzR6CoEtA0OwhCLoEBM0egqBLQNDsIQi6BATNHoKgS0DQ7CEIugQEzR6CoEtA0OwhCLoEBM0egqBLQNDsIQi6BATNHoKgS0DQ7CH8QfQSEDR7CPf6LwFBs4cg6BIQNHsIgi4BQbOHIOgSEDR7CIIuAUGzhwj/vWnag58FBM0eotszBLWFZUZbNUS3ZwhqC8uMtmqIbs8Q1BaWGW3VEN2eIagtLDPaqiG6PUNQW1hmtFVDdHuGoLawzGirhuj27BEyUwiaPUS3Z48w6SJo9hDdniGoLSwzdKohuj1DUFtYZuhUQ3R7hqC2sMzQqYbo9gxBbWGZoVMN0e1ZoYX/IscQNHuIbs9K7X9yoxJXJzt0CFoiBEHFQ3R7hqC2sMg4yIbo9gxBbWGRcZAN0e0ZgtrCIuMgG6LbMwS1hUXGQTZEt2cIaguLjINsiG7PENQWFhkH2RDdniGoLSwyDrIhuj1DUFtYZBxkQ3R7hqC2sMg4yIbo9gxBbWGRcZAN0e0ZgtrCIuMgG6LbMwS1hUXGQTZEt2fPJ+hhZ27we30PqU526BC0REhBQetqbT40w4e7qpMdOgQtEVJO0MNu1LJ++3V/dbJDh6AlQsoJ+rnZDh8b/yR//e7+qD8TgOch0k0jT8C2d8ygAGkJvAa1U+jsNShAWsKm38+NmbOZP6EQj/CAM/jGIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQlBP3aL4Lhy1kkTyoLv7iN52qFnUkTLNTGc7XCzqQJFmrjuVphZ9IEC7XxXK2wM2mChdp4rlbYmTTBQm08VyvsTJpgoTaeqxV2Jk2wUBvP1Qo7kyZYqI3naoWdSRMMkBsEBWkQFKRBUJAGQUEaBAVpEBSkQVCQBkFBGgQFaRAUpEFQkAZBQZqMgjZV9fLz4g+5WulfQpLl0brn3d/neUKq18zHj6paZW+lbodse23rBXz8cXxQfPTxzydo03aoGTrl/ZCrlcOu/VDnOKjn3W/yPMLXH7K2ic9N7p2pux8yGfq5Gd9kEH/8swlqHmi/X01/yNbKx49uoOvZlzilaeXYP2g6h6AXhiz3zhx2q2OWA3PsJ82h9wuOfzZBPVuyqXOh4gwT9Xkr9dvfcwjqD9lvWa6IzlrJKGhTrcd3wSw4/vkE7Qe4cUe7ySDotOJ99lbaH7Ncg3rNNK+/b7JcUPs7k/MU743YMfL4ZxPUTGR2OvN+yNaKKclwUP1WuvNVFkG9Zuru/GgmuIytZFy9Hh0jFxz/ZxO0ybVGOrXSvSOqgKAvmU46/s5055uPH5leKiQt6Nec4vO8wGm6L/lP8eaCzVy85Wsl2+KgR/oU/yWLpDpPFtRrpbbPFEx/3eY1Y45mhqWS30quU1uP9CLpC9JMpzfhZW2lI8sM6jVjXoya4aRzITOXoZWesV7FNNMXJOqzXUtNu5/nmyQ/hd424bwdNVMrZa5BFRP1/bmw65JZida5FotOK/bkm6Mdb1+O2b7q9Jppcn1v67Wyz9XK0Qq68PhzswhIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSIChIg6AgDYKCNAgK0iAoSIOgIA2CgjQICtIgKEiDoCANgoI0CArSIChIg6AgDYKCNP8HxjjuvOcCWAgAAAAASUVORK5CYII=" /><!-- --></p>
<p>We can now observe, that the distribution of signficance values is different than for the random data of data1, where the null hypothesis was true. In data3, true differential expression between the conditions exists. The spike-in proteins are expected to change between the conditions, while most of the background proteins are expected to remain stable between conditions. However, in reality this is not always the case - some background proteins might be changing too due to variations in the experimental conditions during the preparation of the dataset.</p>
<p>Let’s see how RolDE has ranked the spike-in proteins known to be changing between the conditions:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span>(<span class="st">&quot;ups&quot;</span>, RolDE.data3[,<span class="dv">1</span>])</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 21 22 23 24 25 26 27</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [26] 29 30 31 32 33 34 35 36 37 38 39 41 42 43 44 46 47 48 54 57 60 67</span></span></code></pre></div>
<p>Most of the spike-in proteins are located near the top of the result list, as expected.</p>
<p>How many of the proteins remain signifcant at the alpha level of 0.05 after the Bonferroni procedure to adjust for multiple hypothesis testing:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(RolDE.data3[,<span class="dv">4</span>]<span class="sc">&lt;=</span><span class="fl">0.05</span>))</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 74</span></span></code></pre></div>
<p>Findings from the DE analysis can be plotted with the <em>plotFindings</em> function included in the RolDE package:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>?plotFindings</span></code></pre></div>
<div class="r-help-page">
<div class="container">
<table width="100%" summary="page for plotFindings">
<tr>
<td>
plotFindings
</td>
<td style="text-align: right;">
R Documentation
</td>
</tr>
</table>
<h2>
Plot RolDE results
</h2>
<h3>
Description
</h3>
<p>
Plot the findings from longitudinal differential expression analysis with RolDE.
</p>
<h3>
Usage
</h3>
<pre class="r">
plotFindings(file_name = NULL, RolDE_res, top_n, col1 = &quot;blue&quot;, col2 = &quot;red&quot;)
</pre>
<h3>
Arguments
</h3>
<table summary="R argblock">
<tr valign="top">
<td>
<code>file_name</code>
</td>
<td>
<p>
a string indicating the file name in which the results should be plotted. Should have a &quot;.pdf&quot; extension. Default is NULL, no file is created.
</p>
</td>
</tr>
<tr valign="top">
<td>
<code>RolDE_res</code>
</td>
<td>
<p>
the RolDE result object.
</p>
</td>
</tr>
<tr valign="top">
<td>
<code>top_n</code>
</td>
<td>
<p>
an integer or a vector of integers indicating what top differentially expressed features should be plotted. If <code>top_n</code> is a single number, the <code>top_n</code> most differentially expressed feature will be plotted (e.g <code>top_n</code>=1 will plot the most differentially expressed feature). If <code>top_n</code> is a vector of numbers, the differentially expressed features corresponding to top detections within the given range will be plotted (e.g. <code>top_n</code>=seq(1:50) will plot the top 50 differentially expressed features). If more than one feature will be plotted, it is advisable to define a suitable file name in <code>file_name</code>.
</p>
</td>
</tr>
<tr valign="top">
<td>
<code>col1</code>
</td>
<td>
<p>
a string indicating which color should be used for replicates / individualsin in condition 1. The default is blue.
</p>
</td>
</tr>
<tr valign="top">
<td>
<code>col2</code>
</td>
<td>
<p>
a string indicating which color should be used for replicates / individualsin in condition 2. The default is red.
</p>
</td>
</tr>
</table>
<h3>
Details
</h3>
<p>
The function plots the longitudinal expression of the top RolDE findings. The function can plot either the expression of a single finding or multiple top findings as indicated by the <code>top_n</code>. The findings can be plotted into a pdf file as indicated by the <code>file_name</code>. The given <code>file_name</code> should have a &quot;.pdf&quot; extension.
</p>
<h3>
Value
</h3>
<p>
<code>plotFindings</code> Plots the results from the RolDE object.
</p>
<h3>
Examples
</h3>
<pre class="r">
#Plotting the most DE finding. DE results are in the res3 object.
plotFindings(file_name = NULL, RolDE_res = res3, top_n = 1)
</pre>
</div>
</div>
</div>
</div>
<div id="applying-rolde-in-example-data-with-non-aligned-time-points" class="section level2">
<h2>Applying RolDE in example data with non aligned time points</h2>
<p>In addition to data with aligned time points, RolDE can be applied in data with non-aligned time points. However, the <em>aligned</em> parameter in RolDE must now be set to FALSE in order for RolDE to discern that time points in the data are not aligned.</p>
<p>Instead of fixed time points, the integer values in the time point column 3 of the design matrix should now be replaced with continuous numerical values, or time values (e.g. age at the time of sampling).</p>
</div>
<div id="changing-the-settings-for-rolde" class="section level2">
<h2>Changing the settings for RolDE</h2>
<p>While RolDE performs typically very well with the default parameter values, the user might sometimes wish to alter some of these values for a specific kind of analysis. Some of the most important parameters include:</p>
<p>Parameter <em>min_comm_diff</em> controls how many common time points must two replicates (individuals) have in different conditions to be compared. The first value controls the number of common time points for the RegROTS module, while the second one controls the number of common time points for the DiffROTS module. If <em>min_comm_diff</em> is set to “auto”, RolDE will use a value of 3 for the RegROTS module and a value of 1 for the DiffROTS module. In the case of data with non-aligned time points, the first value of <em>min_comm_diff</em> controls how many time values (or similar, e.g. age, temperature) must both replicates (individuals) in different conditions have in the common time interval to be compared.</p>
<p>Defining larger values for the minimum number of common time points to be used in RolDE with data1:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>data1.res<span class="ot">&lt;-</span><span class="fu">RolDE</span>(<span class="at">data=</span>data1, <span class="at">des_matrix=</span>des_matrix1, <span class="at">doPar=</span>T, <span class="at">n_cores=</span><span class="dv">3</span>, <span class="at">min_comm_diff =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">3</span>))</span></code></pre></div>
<p>Using the above values for <em>min_comm_diff</em> requires the replicates in different conditions to have 4 common time points to be compared in the RegROTS module and 3 common time points to be compared in the DiffROTS module.</p>
<p><em>min_feat_obs</em> controls the number of non-missing values a feature must have for a replicate (an individual) in a condition to be compared in the RegROTS module and the DiffROTS module (in data with aligned time points). A feature is required to have at least <em>min_feat_obs</em> non-missing values for both replicates in the different conditions to be compared. If lowered, more missing values are allowed but the analysis may become less accurate. If increased, only replicates (individuals) with less missing values for a feature are compared.</p>
<p>The user can control the degree of polynomials used by the RegROTS and the PolyReg modules via the <em>degtree_RegROTS</em> and the <em>degree_PolyReg parameters</em>. If left to “auto”, RolDE will determine the suitable degrees automatically as described in (Valikangas et al.)</p>
<p>Using RolDE with non default user given polynomial degrees for the RegROTS and PolyReg modules:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>data1.res<span class="ot">&lt;-</span><span class="fu">RolDE_Main</span>(<span class="at">data=</span>data1, <span class="at">des_matrix=</span>des_matrix1, <span class="at">n_cores=</span><span class="dv">3</span>, <span class="at">degree_RegROTS =</span> <span class="dv">2</span>, <span class="at">degree_PolyReg =</span> <span class="dv">3</span>)</span></code></pre></div>
<p>By default, RolDE uses fixed effects only regression with a common intercept and slope for the replicates (individuals) when time points in the data are aligned and mixed effects models with a random effect for the individual baseline (intercept) if the time points are non aligned for the PolyReg and the DiffROTS (only in data with non aligned time points) modules. This behaviour is controlled with the parameter <em>model_type</em> and the default behaviour is induced when <em>model_type</em> is allowed to be “auto”. However, the user can choose to use mixed effects regression modelling when appropriate by setting the parameter <em>model_type</em> as “mixed0” for random effects for the individual baseline and setting <em>model_type</em> as “mixed1” for an individual baseline and slope. Fixed effects only models can be chosen to be used by setting as “fixed”. Valid inputs for  are “auto” (the default), “mixed0”, “mixed1” and “fixed”.</p>
<p>Analyzing data1 using mixed effects modelling with random intercepts for the replicates in the PolyReg module of RolDE:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>data1.res<span class="ot">&lt;-</span><span class="fu">RolDE_Main</span>(<span class="at">data=</span>data1, <span class="at">des_matrix=</span>des_matrix1, <span class="at">n_cores=</span><span class="dv">3</span>, <span class="at">model_type=</span><span class="st">&quot;mixed0&quot;</span>)</span></code></pre></div>
<p>Analyzing data1 using mixed effects modelling with random intercepts and also random linear slopes for the replicates in the PolyReg AND the DiffROTS modules of RolDE:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>data1.res<span class="ot">&lt;-</span><span class="fu">RolDE</span>(<span class="at">data=</span>data1, <span class="at">des_matrix=</span>des_matrix1, <span class="at">n_cores=</span><span class="dv">3</span>, <span class="at">model_type=</span><span class="st">&quot;mixed1&quot;</span>)</span></code></pre></div>
<p>Altering the <em>model_type</em> parameter has an effect for the DiffROTS module only when the time points in the data are non-aligned. In non-aligned time point data, the expression level differences between the conditions in the DiffROTS module is examined when accounting for time-associated trends of varying complexity in the data.</p>
</div>
<div id="preparation-of-data-for-rolde" class="section level2">
<h2>Preparation of data for RolDE</h2>
<p>The data for RolDE needs to be appropriately preprocessed. RolDE does not perform any filtering or normalization for the data; such preprocessing must be performed prior to applying RolDE. Similarly, adjusting for possible confounding effects in the data must be performed before the application of RolDE.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Elo, Laura, Filen S, Lahesmaa R, et al. Reproducibility-optimized test statistic for ranking genes in microarray studies. IEEE/ACM Trans. Comput. Biol. Bioinform. 2008; 5:423-31.</p>
<p>Suomi T, Seyednasrollah F, Jaakkola MK, et al. ROTS: An R package for reproducibility-optimized statistical testing. PLoS Comput. Biol. 2017; 13:5.</p>
<p>Storey JD, Bass AJ, Dabney A, et al. qvalue: Q-value estimation for false discovery rate control. 2019.</p>
<p>Välikangas T, Suomi T, ELo LL, et al. Enhanced longitudinal differential expression detection in proteomics with robust reproducibility optimization regression. bioRxiv 2021.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Session info</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; R Under development (unstable) (2021-11-26 r81252)</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running under: Windows 10 x64 (build 14393)</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] LC_COLLATE=Finnish_Finland.1252  LC_CTYPE=Finnish_Finland.1252   </span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] LC_MONETARY=Finnish_Finland.1252 LC_NUMERIC=C                    </span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [5] LC_TIME=Finnish_Finland.1252    </span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] knitr_1.36   printr_0.2   RolDE_0.99.1</span></span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] highr_0.9                   compiler_4.2.0             </span></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] GenomeInfoDb_1.30.0         XVector_0.34.0             </span></span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] MatrixGenerics_1.6.0        bitops_1.0-7               </span></span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] iterators_1.0.13            tools_4.2.0                </span></span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] rngtools_1.5.2              zlibbioc_1.40.0            </span></span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] digest_0.6.29               nlme_3.1-153               </span></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] evaluate_0.14               lattice_0.20-45            </span></span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] rlang_1.0.2                 doRNG_1.8.2                </span></span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] Matrix_1.3-4                foreach_1.5.1              </span></span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] DelayedArray_0.20.0         cli_3.1.0                  </span></span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] rstudioapi_0.13             yaml_2.2.1                 </span></span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [23] parallel_4.2.0              xfun_0.28                  </span></span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] fastmap_1.1.0               GenomeInfoDbData_1.2.7     </span></span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [27] stringr_1.4.0               S4Vectors_0.32.3           </span></span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [29] IRanges_2.28.0              stats4_4.2.0               </span></span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] grid_4.2.0                  Biobase_2.55.0             </span></span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [33] rmarkdown_2.11              magrittr_2.0.1             </span></span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [35] codetools_0.2-18            matrixStats_0.61.0         </span></span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [37] htmltools_0.5.2             BiocGenerics_0.41.2        </span></span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [39] GenomicRanges_1.46.1        SummarizedExperiment_1.24.0</span></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] stringi_1.7.6               RCurl_1.98-1.5</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
